app/uk/gov/hmrc/tai/model/hip/reads/OtherNonTaxCodeIncomeHipReads.scala

```diff
@@ -14,29 +14,27 @@
  * limitations under the License.
  */
 
-package uk.gov.hmrc.tai.model.domain.income
+package uk.gov.hmrc.tai.model.hip.reads
 
-import play.api.libs.json._
-import uk.gov.hmrc.tai.model.domain._
+import play.api.libs.json.*
+import uk.gov.hmrc.tai.model.domain.*
+import uk.gov.hmrc.tai.model.domain.income.*
 
 object OtherNonTaxCodeIncomeHipReads {
-  implicit val format: Format[OtherNonTaxCodeIncome] = Json.format[OtherNonTaxCodeIncome]
 
-  val otherNonTaxCodeIncomeReads: Reads[Seq[OtherNonTaxCodeIncome]] = (json: JsValue) => {
-    val extractedIabds: Seq[NpsIabdSummary] =
-      json.as[Seq[NpsIabdSummary]](NpsIabdSummaryHipReads.iabdsFromTotalLiabilityReads)
-    JsSuccess(nonTaxCodeIncomes(extractedIabds))
-  }
+  implicit val format: OFormat[OtherNonTaxCodeIncome] = Json.format[OtherNonTaxCodeIncome]
 
-  private def nonTaxCodeIncomes(iabds: Seq[NpsIabdSummary]): Seq[OtherNonTaxCodeIncome] =
-    iabds collect {
-      case iabd if nonTaxCodeIncomesMap.isDefinedAt(iabd.componentType) =>
-        OtherNonTaxCodeIncome(
-          nonTaxCodeIncomesMap(iabd.componentType),
-          iabd.employmentId,
-          iabd.amount,
-          iabd.description
-        )
+  val otherNonTaxCodeIncomeReads: Reads[Seq[OtherNonTaxCodeIncome]] =
+    NpsIabdSummaryHipReads.iabdsFromTotalLiabilityReads.map { iabds =>
```

Comment:
This depends on another class. Would be better if self-contained. Also not as efficient as it could be since it reads everything in then effectively filters

Reply:
Please note there are 2 case classes for the same iabd details :( 
IabdDetails and NpsIabdRoot